src/Workspaces/Core/Portable/SourceGeneratorTelemetry/SourceGeneratorTelemetryCollectorWorkspaceService.cs

```diff
@@ -61,25 +62,36 @@ public void CollectRunResult(
         }
     }
 
-    public void ReportStatisticsAndClear(FunctionId functionId)
+    public ImmutableArray<ImmutableDictionary<string, object?>> FetchKeysAndAndClear()
     {
+        var arrayBuilder = ImmutableArray.CreateBuilder<ImmutableDictionary<string, object?>>();
```

Comment:
```suggestion
        var arrayBuilder = new FixedSizeImmutableArrayBuilder<ImmutableDictionary<string, object?>>(_elapsedTimeByGenerator.Count);
```

or just .SelectAsArray